lib/ruby_lsp/node_context.rb

```diff
@@ -17,5 +17,10 @@ def initialize(closest, parent, nesting)
       @parent = parent
       @nesting = nesting
     end
+
+    sig { returns(String) }
+    def fully_qualified_name
+      @nesting.join("::")
```

Comment:
I wonder if we should set this in the `initialize` and add an `attr_reader` for it, since the nesting doesn't change after instantiation. Thoughts?